src/sinks/util/buffer/metrics.rs

```diff
@@ -148,23 +144,22 @@ impl<N: MetricNormalize> Batch for MetricsBuffer<N> {
         if self.num_items() >= self.max_events {
             PushResult::Overflow(item)
         } else {
-            if let Some(item) = N::apply_state(&mut self.state, item.into_metric()) {
-                let new_entry = match item.data.kind {
-                    MetricKind::Absolute => MetricEntry(item),
-                    MetricKind::Incremental => {
-                        // Incremental metrics update existing entries, if present
-                        let entry = MetricEntry(item);
-                        self.metrics
-                            .take(&entry)
-                            .map(|mut existing| {
-                                existing.data.update(&entry.data);
-                                existing
-                            })
-                            .unwrap_or(entry)
-                    }
-                };
-                self.metrics.replace(new_entry);
-            }
+            let item = item.into_metric();
+            let new_entry = match item.data.kind {
+                MetricKind::Absolute => MetricEntry(item),
+                MetricKind::Incremental => {
+                    // Incremental metrics update existing entries, if present
+                    let entry = MetricEntry(item);
+                    self.metrics
+                        .take(&entry)
+                        .map(|mut existing| {
+                            existing.data.update(&entry.data);
+                            existing
+                        })
```

Comment:
```suggestion
                        .for_each(|mut existing| {
                            existing.data.update(&entry.data);
                        })
```

Reply:
Unless I mistyped it, that doesn't work. The return from `metrics.take` is an `Option` not an iterator.

Reply:
Whoops, missed that this is an `Option` 😄